tests/integration/test_schema_protobuf.py

```diff
@@ -1065,6 +1056,15 @@ async def test_reference_update_creates_new_schema_version(testcase: ReferenceTe
     assert references[0].get("subject") == "wr_s1"
     assert references[0].get("version") == 2
 
+    # Assert when querying the schema id with first schema version with references correct schema id is returned.
+    body = {
+        "schemaType": test_schemas[1].schema_type,
+        "schema": test_schemas[1].schema_str,
+        "references": test_schemas[1].references,
+    }
+    res = await registry_async_client.post("subjects/wr_s2", json=body)
+    assert res.json_result.get("id") == 2
```

Comment:
How about instead first retrieving the id of the schema and comparing it here instead of hardcoding its id?

Reply:
I changed to a loop where each created schema is tested to return correct id when queried with schema string and references.